### PR TITLE
feat(rust): added timeout option for send command when trying to acce…

### PR DESCRIPTION
**Pull request by michealkeines**
## Added

- Added option to set custom time out for send command
- Added `if node exist check` before sending the message

## Output

error handling

## Notes

without timeout option, the default timeout is set to 30 secs

```rust
/// A default timeout in seconds
pub const DEFAULT_TIMEOUT: u64 = 30;
```

## Checks

- [x] All commits in this Pull Request are [signed](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits) and Verified by Github.
- [x] All commits in this Pull Request follow the Ockam [commit message convention](https://github.com/build-trust/.github/blob/main/CONTRIBUTING.md#commit-messages).
- [x] I accept the Ockam Community [Code of Conduct](https://github.com/build-trust/.github/blob/main/CODE_OF_CONDUCT.md).
- [x] I have accepted the Ockam [Contributor License Agreement](https://github.com/build-trust/ockam-contributors/blob/main/CLA.md) by adding my Git/Github details in a row at the end of the [CONTRIBUTORS.csv](https://github.com/build-trust/ockam-contributors/blob/main/CONTRIBUTORS.csv) file in a separate pull request to the [build-trust/ockam-contributors](https://github.com/build-trust/ockam-contributors) repository.

### CODE REVIEW

File: implementations/rust/ockam/ockam_node/src/context.rs [hunk 1]
```diff
@@ -427,6 +427,27 @@ impl Context {
...
+    /// [`receive`]: Self::receive
+    pub async fn send_and_receive_with_timeout<R, M, N>(&self, route: R, msg: M, timeout: u64) -> Result<N>
+    where
+        R: Into<Route>,
+        M: Message + Send + 'static,
+        N: Message,
+    {
+        let mut child_ctx = self.new_detached(Address::random_local()).await?;
+        child_ctx.send(route, msg).await?;
+        let duration = std::time::Duration::from_secs(timeout);
```
Comment (mrinalwadhwa) [inline implementations/rust/ockam/ockam_node/src/context.rs]:
```suggestion
        let duration = core::time::Duration::from_secs(timeout);
```

Comment (mrinalwadhwa) [reply implementations/rust/ockam/ockam_node/src/context.rs]: This is why our no_std check failed https://github.com/build-trust/ockam/runs/7914079845?check_suite_focus=true#step:7:15

Comment (michealkeines) [reply implementations/rust/ockam/ockam_node/src/context.rs]: I have updated it with core

**Comment by mrinalwadhwa** [comment]
@michealkeines awesome! thank you for sending this.

A couple of suggestions:
1. `cargo fmt` will take care of the formatting check [failing](https://github.com/build-trust/ockam/runs/7914078806?check_suite_focus=true)
2. `cargo clippy` is unhappy, have a look at the error [here](https://github.com/build-trust/ockam/runs/7914078376?check_suite_focus=true) 
3. the commit linter is failing because the commit message is too long in d560f47571a54ff798a67846eb4b98f1e60aeaca, could you squash the two commits into one and move some of the description into the message body. I can do this last part for you during the merge as well.

**Comment by michealkeines** [comment]
I ran the following

`cargo fmt --all`

but `cargo clippy` gave me a lot of errors (i am running it in m1 arm)

image

**Comment by mrinalwadhwa** [comment]
From the screenshot it looks like the compiler wants you to run `cargo clean && cargo build`... did you try that?

**Comment by michealkeines** [comment]
image

This should work now, i hope

**Review by mrinalwadhwa — APPROVED** [review]

**Comment by mrinalwadhwa** [comment]
Thank you for another awesome contribution! 🥳 🎈

**Pull request merged**